src/safir/github/models.py

```diff
@@ -0,0 +1,372 @@
+    """A Pydantic model for the ``user`` field found in GitHub API resources.
+
+    This contains brief (public) info about a user.
+    """
+
+    login: str = Field(title="Login name", description="GitHub username.")
+
+    html_url: HttpUrl = Field(
+        title="Profile URL", description="Homepage for the user on GitHub."
+    )
+
+    url: HttpUrl = Field(
+        title="API URL",
+        description="URL for the user's resource in the GitHub API.",
+    )
+
+    avatar_url: HttpUrl = Field(
+        title="Avatar image URL", description="URL to the user's avatar."
+    )
+
+
+class GitHubRepositoryModel(BaseModel):
+    """A Pydantic model for the ``repository`` field, often found in webhook
+    payloads.
+
+    https://docs.github.com/en/rest/repos/repos#get-a-repository
+    """
+
+    name: str = Field(
+        title="Repository name",
+        description="Excludes owner prefix.",
+        example="times-square-demo",
+    )
+
+    full_name: str = Field(
+        title="Full name",
+        description=(
+            "Full name, including owner prefix "
+            "(e.g. ``lsst-sqre/times-square-demo``).)"
+        ),
+        example="lsst-sqre/times-square-demo",
+    )
+
+    owner: GitHubRepoOwnerModel = Field(description="The repository's owner.")
+
+    default_branch: str = Field(
+        description="The default branch (e.g. main).", example="main"
+    )
+
+    html_url: HttpUrl = Field(
+        description="URL of the repository for browsers.",
+        example="https://github.com/lsst-sqre/times-square-demo",
+    )
+
+    branches_url: str = Field(
+        description="URI template for the repo's branches endpoint.",
+        example=(
+            "https://github.com/lsst-sqre/times-square-demo/branches{/branch}"
+        ),
+    )
+
+    contents_url: str = Field(
+        description="URI template for the contents endpoint.",
+        example=(
+            "https://github.com/lsst-sqre/times-square-demo/contents/{+path}"
+        ),
+    )
+
+    trees_url: str = Field(
+        description="URI template for the Git tree API.",
+        example=(
+            "https://github.com/lsst-sqre/times-square-demo/git/trees{/sha}"
+        ),
+    )
+
+    blobs_url: str = Field(
+        description="URI template for the Git blobs API.",
+        example=(
+            "https://github.com/lsst-sqre/times-square-demo/git/blobs{/sha}"
+        ),
+    )
+
+
+class GitHubPullState(str, Enum):
+    """The state of a GitHub PR.
+
+    https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request
+    """
+
+    open = "open"
+    closed = "closed"
+
+
+class GitHubPullRequestModel(BaseModel):
+    """A Pydantic model for a GitHub Pull Request.
+
+    This is also the ``pull_request`` field inside the
+    `~safir.github.webhooks.GitHubPullRequestEventModel`.
+
+    https://docs.github.com/en/rest/pulls/pulls#get-a-pull-request
+    """
+
+    html_url: HttpUrl = Field(description="Web URL of the PR.")
```

Comment:
It feels weird to have `description` and not `title`, but given that we're never using these models in an OpenAPI schema (where that would matter), it doesn't seem to matter. I admit I would be tempted to add `title` just in case for the future, but I would probably just be making tedium for myself.